clients/fides-js/src/fides.ts

```diff
@@ -120,15 +180,58 @@ const init = async ({
   geolocation,
   options,
 }: FidesConfig) => {
+  const { fides_user_device_id: fidesUserDeviceId } =
+    getOrMakeFidesCookie().identity;
```

Comment:
I think this means we need to rethink a bit how we initialize that cookie here. The `getOrMakeFidesCookie()` function is a two-parter (the hint is in the name!) that will either:
* get the cookie, if it exists
* make a new cookie based on the consent defaults

Given that we can't rely on having the consent defaults at startup anymore, we likely need to split this up into something more like:
```typescript
// Read existing cookie, if one exists
let fidesCookie = getExistingFidesCookie();

// Generate a device ID (or use the one from the Fides cookie)
const fidesUserDeviceID = // ...

// Load the existing user preferences
const existingPreferences = fidesCookie?.consent;

// Geolocate the local device (or use the provided geolocation)
// ...

// Fetch the experience (or use the provided experience)
// ...

// Initialize the overlay (or not)
// ...

// Apply GPC preferences
// ...

// Initialize _Fides object
// ...

// Save the Fides cookie
saveFidesCookie({ ... });

// Emit an "onInit" event (or similar)
// ...

// Done!
```

Reply:
I've updated `getOrMakeFidesCookie` to take an explicit `device_id` param, which will use that instead of generating a new one. So I do this the 2nd time I call the method on L:229:

```
// Load any existing user preferences from the browser cookie
  const cookie = getOrMakeFidesCookie(
    consentDefaults,
    fidesUserDeviceId,
    options.debug
  );
```

Will that work, or you still think I should refactor as you mentioned above?